app/services/CartService.java

```diff
@@ -0,0 +1,75 @@
+package services;
+
+import io.sphere.sdk.carts.Cart;
+import io.sphere.sdk.products.ProductProjection;
+import io.sphere.sdk.products.ProductVariant;
+import pactas.models.PactasContract;
+import pactas.models.PactasCustomer;
+import play.mvc.Http;
+
+import java.util.Optional;
+
+/**
+ * Business service that provides access to the SPHERE.IO Cart API.
+ */
+public interface CartService {
+
+    /**
+     * If there is no {@code SessionKeys.CART_ID} in the existing session, a new {@code Cart} will be created in the
+     * Sphere backend and its {@code SessionKeys.CART_ID} will be stored in the users {@code Http.Session session}.
+     * Otherwise it retrieves an existing Cart via its {@code SessionKeys.CART_ID}, but never null.
+     *
+     * @param session the Play session object, must not be null
+     * @return the newly created, or fetched {@code Cart} object, must not be null
+     */
+    Cart getOrCreateCart(Http.Session session);
+
+    /**
+     * Resets an existing {@code Cart}.
+     * It performs two API calls, one to remove the Carts LineItems and another one to clear the {@code CustomObject}
+     * that holds the selected frequency of the product.
+     *
+     * @param cart the {@code Cart} object to clear, must not be null
+     * @return the updated {@code Cart} object as result from the update command to Sphere API, must not be null
+     */
+    Cart clearCart(final Cart cart);
+
+
+    /**
+     * Adds the user selected {@code ProductVariant} of the {@code ProductProjection} with a given frequency to the
+     * users {@code Cart}.
+     *
+     * @param cart the user's current {@code Cart} object, must not be null
+     * @param product the {@code ProductProjection} to add, must not be null
+     * @param variant the selected {@code ProductVariant}, must not be null
+     * @param frequency the selected delivery frequency
+     */
+    void setProductToCart(final Cart cart, final ProductProjection product, final ProductVariant variant, final int frequency);
+
+    /**
+     * Returns the value of the {@code CustomObject} named ShopKeys.FREQUENCY, that is bound to the {@code Cart}
+     * with the given cardId.
+     *
+     * @param cartId the identifier of the {@code Cart} object, the {@code CustomObject} is bound to
+     * @return the value of the {@code CustomObject}. If no {@code CustomObject} found, it returns 0
+     */
+    int getFrequency(final String cartId);
```

Comment:
This is a minor issue, but I think since all other methods of the interface are expecting a `Cart`, there's no reason to expect the `Cart ID` here (at least none that I could find). Anyway with the custom attributes in line items this will be changed :)